### ISSUE TITLE
Items added to the DOM by javascript profiles are not shown on screenshots

### ISSUE THREAD

**Issue by iodbh** [scrapinghub/splash#507]
Hello - and first of all congratulations on a amazing job on Splash.

I ran into a issue where elements injected into the DOM of the target page do appear in the HTML of the page, but not on screenshots.

The elements are injected via javascript scripts that are loaded using the "profiles" feature.

Here are more details about my situation :

 - I'm using the docker distribution of splash, with the latest version pulled from scrapinghub's repo at the time of writing this
 - I'm on OSX
 - I'm starting Slash with the following command : `docker run --volume=/local/path/to/profiles:/profiles -p 5023:5023 -p 8050:8050 -p 8051:8051 scrapinghub/splash --js-cross-domain-access --js-profiles-path=/profiles`
- calls to `render.html` or `render.js` do show the elements injected by the javascript, but they don't appear in screenshots obtained vi a`render.png` or `render.jpeg`

The code i'm having the issue with

```python
import requests, json
import shutil

SPLASH_BASE_URL = 'http://127.0.0.1:8050/render.png'
SPLASH_BASE_URL_JSON = 'http://127.0.0.1:8050/render.json'
OUTPUT_PATH = '/tmp/test_underline_image.png'

headers = {'Content-Type': 'application/json'}

# construct POST data
args = {"url": "http://test.iodbh.net",
        "js": "generic",
        "wait": 10, # i added a long wait to make sure this isn't a timing issue
        "render_all": 1,
        "console": 1,
        "html": 1,
        }

# this request grabs the screenshot
r = requests.post(SPLASH_BASE_URL, stream=True, data=json.dumps(args),
                  headers=headers)
print(r.request.body)

# this second request is here to get the page's html and console output
r2 = requests.post(SPLASH_BASE_URL_JSON, data=json.dumps(args), headers=headers)
print(r2.json())
print(r2.json()['html'])
console_output = r2.json().get('console', None)
print("[BEGIN CONSOLE OUTPUT]")
if console_output is not None:
    for line in console_output:
        print(line)
else:
    print('[!] no js console output')
    print("[END CONSOLE OUTPUT]")

if r.status_code == 200:
    # download the image
    with open(OUTPUT_PATH, 'wb') as f:
        r.raw.decode_content = True
        shutil.copyfileobj(r.raw, f)
else:
    print(r.text)
```

here is the structure of my `profiles` directory:

```
generic/
    100-jquery-3.1.0.min.js     # regular distribution of jquery
    101-jquery-noconflict.js    # see below
    201-inject-css.js                # see below
    301-hint-css-tooltips.js    # see below
```

__101-jquery-noconflict.js__ : 

this simply makes sure jquery does not conflict with another version present on the same page :

```javascript
var l$ = $.noConflict(true);
```

__201-inject-css.js__ :

this adds the CSS i'm using into the page : 

```javascript
l$('head').append('<link rel="stylesheet" href="http://test.iodbh.net/libs/hint.min.css">');
```

___301-hint-css-tooltips.js__ :

```javascript
console.log('injecting hint.css...');
l$('p').append('<a class="hint--bottom  hint--always" aria-label="FUCK YEAH TOOLTIP !">You can also make tooltips...</a>');
console.log('done.');
```

I've run into this issue with other requests, this is simply the most basic case in which i have that problem. I should add that while some modifications are not reflected in the screenshot (injecting elements), others are (such as adding other elements to other pages, or calls to jquery's `css()` method).

I am available to provide any additional information, or look into the code if you point me into the right direction.

**Comment by iodbh** [comment]
For clarity, here is a sample response from `render.json` that shows a screenshot without the extra elements while they are present in the html source :

[https://gist.github.com/iodbh/d4e521d1a043fece6b2011ebe528a1de](url)

**Comment by Gallaecio** [comment]
Did you ever manage to solve your issue?

**Comment by iodbh** [comment]
No, but I haven't tried doing this again since shortly after opening this issue.

**Comment by Gallaecio** [comment]
Shall we close this issue, or is this something you plan to go back to?

**Comment by iodbh** [comment]
this is far from critical for me but if the issue is with splash and not my code it is probably worth looking into as it can break some use cases (in this instance taking "annotated" screenshots of pages). Ultimately it caused me to use a different headless browser for that project.
